### ISSUE TITLE
Segfault with 6.4.5, 6.7.5

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#877]
**Reported by shiki08 on 27 Apr 39395801 04:26 UTC**
First, I installed 'boinc' from community repo. It wasnt working at all. then I tried to build them both (6.4.5, 6.7.5). First, I failed with 6.4.5. (Used Arch linux ABS system). Then downloaded it from SVN (tagged), tried to compile that way. (Disabled the gui for faster speed (cuz X sucks :))).

That way it compiles. BUT. I launch the 'boinc_client' and I get this (I get the same when I try to run 6.4.5 !!! ..I can paste that too):

```
[~](shiki@ayanami)$ boinc
Starting BOINC client version 6.7.5 for x86_64-pc-linux-gnu
.
.
.
24-May-2009 09:07:00 [Version change (6.4.5 -> 6.7.5)
24-May-2009 09:07:00 [---](---]) No general preferences found - using BOINC defaults
24-May-2009 09:07:00 [Preferences limit memory usage when active to 1005.35MB
24-May-2009 09:07:00 [---](---]) Preferences limit memory usage when idle to 1809.63MB
SIGSEGV: segmentation violation
Segmentation fault
```

gcc (GCC) 4.4.0

http://pastebin.com/f517f63e9 (Package versions)

system: Arch Linux x86-64
/etc/make.conf: (Tried to remove the make.conf, no use)
CFLAGS="-march=core2 -mtune=core2 -fomit-frame-pointer -mmmx -msse -msse2 -msse3  -O2 -pipe
CXXFLAGS="${CFLAGS}

MAKEOPTS="-j6"
(I made MAKEOPTS get ignored , because it wont compile like that :( )

If you need any info, I'll answer ASAP.
\ Sorry for setting prior BLOCKER, but it is really a blocker if I cant even start it. \

**Comment by romw** [comment]
**Commented by shiki08 on 16 Jan 39395892 18:13 UTC**
How could I post a debug info? Compiled it with -debug flag, but ... strace gives 8MB or 180mb file... GDB? Something?

**Comment by romw** [comment]
**Commented by Nicolas on 2 Dec 39396889 08:26 UTC**
Not a blocker if you're the only one (so far) with the problem.

**Comment by romw** [comment]
**Commented by Nicolas on 2 Mar 39396894 06:40 UTC**
Remove ```-O2``` from the flags and add ```-g```. Recompile.

Run ```gdb boinc``` and type ```run``` in the ```(gdb)``` prompt. Once it crashes, type ```bt```.

**Comment by romw** [comment]
**Commented by shiki08 on 27 Nov 39398987 12:00 UTC**
Nicolas > I will be at home from Friday, so I'll recompile and answer when I get back. By the way, a lot of users had the same problem on IRC, but no one would post a bugreport to the upstream. 

Tiffanyand > WTF?

**Comment by romw** [comment]
**Commented by Nicolas on 21 Oct 39399420 22:13 UTC**
Replying to [shiki08](comment:6):
> Nicolas > I will be at home from Friday, so I'll recompile and answer when I get back. By the way, a lot of users had the same problem on IRC, but no one would post a bugreport to the upstream. 
What IRC channel?

> Tiffanyand > WTF?
It was spam.

**Comment by romw** [comment]
**Commented by shiki08 on 9 Jan 39407580 12:26 UTC**
Added what you asked (btw it was Arch linux IRC):

checking build system type... x86_64-unknown-linux-gnu/build/boinc/boinc_core_rel
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for a BSD-compatible install... /bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking for C compiler default output file name... 
configure: error: in `/home/shiki/Documents/build/boinc/boinc_core_release_6_7_5':
configure: error: C compiler cannot create executables
See `config.log' for more details.

**Comment by romw** [comment]
**Commented by shiki08 on 26 Sep 39407588 09:20 UTC**
CFLAGS="-march=core2 -mtune=core2 -ffast-math -fomit-frame-pointer -mmmx -msse -msse2 -msse3 -CXXFLAGS="${CFLAGS} ./configure

This way it configures, (without --enable-optimize).

Now, it doesnt compiles:
http://pastebin.com/f5cbbb448

Any idea?

**Comment by romw** [comment]
**Commented by Nicolas on 17 Mar 39407679 08:53 UTC**
You have your quote marks wrong; it's passing "-CXXFLAGS=" as one of the CFLAGS! Compare:

CFLAGS="-march=core2 -mtune=core2 -ffast-math -fomit-frame-pointer -mmmx -msse -msse2 -msse3 -CXXFLAGS="${CFLAGS} ./configure

CFLAGS="-march=core2 -mtune=core2 -ffast-math -fomit-frame-pointer -mmmx -msse -msse2 -msse3'''"''' CXXFLAGS="${CFLAGS}" ./configure

**Comment by romw** [comment]
**Commented by shiki08 on 7 Dec 39407866 04:26 UTC**
No difference... oO

http://pastebin.com/f4c5f05af

Disabled --fast-math:

http://pastebin.com/f472d9e84

The error exactly (as you can see it):
../lib/str_util.h:41: error: new declaration &#8216;char* strcasestr(const char*, const char*)&#8217;
/usr/include/string.h:367: error: ambiguates old declaration &#8216;const char* strcasestr(const char*, const char*)&#8217;

**Comment by romw** [comment]
**Commented by shiki08 on 24 Feb 39407869 08:53 UTC**
This was 6.7.5 as you can see it. Should I get the NEWEST version (from SVN) , or I should keep trying with the 6.4.5 ?

**Comment by romw** [comment]
**Commented by Nicolas on 21 Jun 39407885 00:26 UTC**
You were initially reporting a segfault. Now you're having build errors; totally different problem.

Try to build 6.4.5 and see if you can reproduce the segfault. If you can't even compile it, please open a ''separate ticket''.

**Comment by romw** [comment]
**Commented by shiki08 on 5 Aug 39408052 16:26 UTC**
Okay, I was able to compile and run 6.4.5.

First of all, some BOINC install (I suppose it was the 6.7.5 version) made my partition dirty, so I just removed all files by "sudo updatedb && locate -i 'boinc'". By hand. :)

Then did this:
1) Got source from SVN
2) patch -p0 < gcc4.4.0.patch
3) ./_automake
4) CFLAGS="-march=core2 -mtune=core2 -mmmx -msse -msse2 -msse3 -g" CXXFLAGS="${CFLAGS}" ./configure --prefix=/usr
5) make
6) sudo make install

Now it works.. SO FAR (and I cant make libcudart.so thingie work). Ticket can be closed. Reopening a new one with the 6.7.5 compiling issue (if there is no such ticket so far. Thanks for your help.)
